### PR TITLE
feat(baseCheckbox): adds density support for Checkbox, Radio components.

### PR DESCRIPTION
**Pull request by stephanLeece**
# Purpose of PR

This PR adds density support to allow for decreased font-size and line height when the component is wrapped in a density container. Adds examples for the Checkbox and Radio, to the DensityContainer storybook stories

Ticket: https://contentful.atlassian.net/browse/CFCX-956

## PR Checklist

- [x] I have read the relevant `readme.md` file(s)
- [x] All commits follow our [Git commit message convention](https://github.com/contentful/forma-36/tree/main/packages/forma-36-react-components#commits)
- [x] Tests are added/updated/not required
- [x] Tests are passing
- [x] Storybook stories are added/updated/not required
- [x] Usage notes are added/updated/not required
- [x] Has been tested based on [Contentful's browser support](https://www.contentful.com/faq/about-contentful/#which-browsers-does-contentful-support)
- [x] Doesn't contain any sensitive information

### CODE REVIEW

File: packages/components/forms/src/BaseCheckbox/types.ts
Comment (cf-remylenoir) [inline packages/components/forms/src/BaseCheckbox/types.ts:51]: You should be able to replace this with the Density type.

https://github.com/contentful/forma-36/blob/408cd9aadf665d06f5b06856b6cea7ffa6d4c4e8/packages/components/utils/src/useDensity/useDensity.ts?plain=1#L3

File: packages/components/forms/src/BaseCheckbox/BaseCheckbox.styles.ts
Comment (cf-remylenoir) [inline packages/components/forms/src/BaseCheckbox/BaseCheckbox.styles.ts:29]: We could define the default (low) value here too, based on the design spec, as you did for the `BaseCheckbox` Text component.

Comment (stephanLeece) [reply packages/components/forms/src/BaseCheckbox/BaseCheckbox.styles.ts:29]: good shout, couldn't hurt eh?

Comment (stephanLeece) [reply packages/components/forms/src/BaseCheckbox/BaseCheckbox.styles.ts]: done :)

**Comment by cf-remylenoir** [comment]
@damann happy to hear your feedback on the addition of the default values (based on your [design spec](https://www.figma.com/file/IZfaouOHnhiX8OVbVAAPzx/F36-911-High-density?type=design&node-id=116-1560&mode=design&t=zmvNxxorR1XlYSCD-0)). 

The Chromatic changes show how much our baselines changed from this update. 
I am happy to approve them, but I would love to get your approval too.

Previously, no default values were assigned to the elements and they were inherited from the global values (i.e. line-height-default': '1.5').

**Review by cf-remylenoir — COMMENTED** [review]
Looks good to me, could you please add a changeset too ([see](https://github.com/contentful/forma-36/pull/2644#issuecomment-1881582344)), since the components' API was affected?

Happy to approve after :)

**Comment by stephanLeece** [comment]
> Looks good to me, could you please add a changeset too ([see](https://github.com/contentful/forma-36/pull/2644#issuecomment-1881582344)), since the components' API was affected?
> 
> Happy to approve after :)

Thanks! Added as a patch, based on example here https://github.com/contentful/forma-36/pull/2620/files as they are of similar scope

File: .changeset/wild-wombats-change.md
Comment (cf-remylenoir) [inline .changeset/wild-wombats-change.md:5]: I'd love to make the message more explicit, what about: 

```suggestion
Add high-density support to Checkbox and Radio components.
```

**Review by cf-remylenoir — APPROVED** [review]
Thanks! Once the Chromatic baseline snapshots are accepted by @damann / myself, we can proceed with the merge 👍

File: .changeset/wild-wombats-change.md
Comment (cf-remylenoir) [reply .changeset/wild-wombats-change.md:5]: Ah maybe the revert put the previous message back 👀

Comment (stephanLeece) [reply .changeset/wild-wombats-change.md]: Good shout, updated!

**Review by damann — CHANGES_REQUESTED** [review]
It appears like the distance between checkboxes/radio buttons in the group is smaller than in the design specs.
Screenshot 2024-01-10 at 15 25 37

**Comment by stephanLeece** [comment]
> It appears like the distance between checkboxes/radio buttons in the group is smaller than in the design specs. Screenshot 2024-01-10 at 15 25 37

Is that something we should resolve with design tweaks to the stack component? 

As i've only made changes to the font-size and line-height, i'm not sure how we could fix this with changes to the individual checkbox/radio components...

**Comment by stephanLeece** [comment]
@damann based on storybook, it seems like they are set to have a 0.5rem vertical gap?

Screenshot 2024-01-10 at 15 34 53

**Comment by stephanLeece** [comment]
Okidoki, i've narrowed down the spacing / line height issue. 

Screenshot 2024-01-10 at 15 58 19

`lineHeightCondensed` corresponds to 1.25, which differs from the current default line height. 

`lineHeightM` corresponds to 1.25rem, which should match the current line height. 

I'll go ahead and update in my code.

File: packages/components/forms/package.json
Comment (cf-remylenoir) [inline packages/components/forms/package.json:10]: I was too late to recommend installing this as dev-dependency instead since it is not required for the package to be functional?

**Pull request merged**
